src/main/java/uk/ac/cam/cl/dtg/segue/api/QuestionFacade.java

```diff
@@ -353,4 +353,59 @@ public Response answerQuestion(@Context final HttpServletRequest request,
             return error.toResponse();
         }
     }
+
+    /**
+     * Convert a possible answer into a question specification.
+     *
+     * @param request
+     *            - the servlet request so we can find out if it is a known user.
+     * @param jsonAnswer
+     *            - answer body which will be parsed as a Choice and then converted to a ChoiceDTO.
+     * @return Response containing a QuestionValidationResponse object or containing a SegueErrorResponse .
+     */
+    @POST
+    @Path("generateSpecification")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Turn an answer into a question specification.")
+    public Response generateSpecification(@Context final HttpServletRequest request, final String jsonAnswer) {
```

Comment:
I was wondering about the difference between the specification generated here and what the student submits / what the student's answer actually turns into later. Would there be value in aligning these or converting what the student submits earlier in the API flow?

Reply:
If I've understood you correctly, then that might make sense except that: what the student submits can be _checked_ against the specification but can't be turned into something similar to the specification for reasons. So I think the current shape is correct.

(Reasons are: a line can be counted as going through multiple sectors at any point, so the structure of the user's line is different to the structure of the specification.)